packages/cadl-ranch/src/scenarios-resolver.ts

```diff
@@ -31,16 +31,20 @@ export async function findScenarioCadlFiles(scenariosPath: string): Promise<Cadl
   const pattern = [`${normalizedScenarioPath}/**/client.tsp`, `${normalizedScenarioPath}/**/main.tsp`];
   logger.debug(`Looking for scenarios in ${pattern}`);
   const fullScenarios = await findFilesFromPattern(pattern);
+  logger.info(`Found ${fullScenarios.length} full scenarios.`);
   const scenarioSet = new Set(fullScenarios);
   const scenarios = fullScenarios.filter((scenario) => {
     // Exclude main.tsp that have a client.tsp next to it, we should use that instead
-    return scenario.endsWith("/main.tsp") && scenarioSet.has(normalizePath(join(dirname(scenario), "client.tsp")));
+    return (
```

Comment:
why was this not working before with building the path. Doing a .replace is less reliable if you end up with something in the path above it would change it.

Reply:
This is not working in my another laptop which is a windows system yesterday. I changed it to use path.sep, but I am using mac now. will not go to the office in the rest week. so I am not sure if this change works for windows or not